frontend/packages/console-dynamic-plugin-sdk/src/extensions/horizontal-nav-tabs.ts

```diff
@@ -28,6 +28,23 @@ export type HorizontalNavTab = ExtensionDeclaration<
   }
 >;
 
+/** Contributes a tab into an HorizontalNav component matching the sectionId. */
+export type NavTab = ExtensionDeclaration<
+  'console.tab',
+  {
+    /** ContextId assigned to the horizontal nav in which the tab will be injected */
+    contextId: string;
+    /** The page to be show in horizontal tab. It takes tab name as name and href of the tab */
+    page: {
+      name: string;
+      href: string;
+    };
+    /** Tab content component. */
+    component: CodeRef<React.ComponentType<PageComponentProps>>;
```

Comment:
Is `PageComponentProps` correct here? TBH I don't know what some of the props do for the horizontal nav tabs extension.